src/connectionController.ts

```diff
@@ -416,58 +422,55 @@ export default class ConnectionController {
     void vscode.window.showInformationMessage('MongoDB connection successful.');
 
     this._activeDataService = newDataService;
-    this._activeConnectionModel = connectionModel;
+    this._activeConnectionModel = await convertConnectionInfoToModel({ connectionOptions });
     this._currentConnectionId = connectionId;
     this._connecting = false;
     this._connectingConnectionId = null;
     this.eventEmitter.emit(DataServiceEventTypes.CONNECTIONS_DID_CHANGE);
     this.eventEmitter.emit(DataServiceEventTypes.ACTIVE_CONNECTION_CHANGED);
 
     // Send metrics to Segment
-    this.sendTelemetry(newDataService, connectionModel, connectionType);
+    this.sendTelemetry(newDataService, connectionType);
 
     return {
       successfullyConnected: true,
       connectionErrorMessage: ''
     };
   }
 
-  async connectWithConnectionId(connectionId: string): Promise<boolean> {
-    if (!this._connections[connectionId]) {
-      throw new Error('Connection not found.');
-    }
+  private _endPrevConnectAttempt (attempt: {
+    connectionId: string,
+    connectingAttemptVersion: null | string,
+    newDataService: DataService
+  }): Boolean {
```

Comment:
```suggestion
  }): boolean {
```

(surprised that TS didn’t error on this one, maybe it’s because we’re not using strict mode in vscode?)